lnwallet/interface.go

```diff
@@ -662,6 +662,14 @@ func InternalKeyForAddr(wallet WalletController, netParams *chaincfg.Params,
 
 	pubKeyAddr, ok := walletAddr.(waddrmgr.ManagedPubKeyAddress)
 	if !ok {
+		// Imported addresses do not provide private keys, so they
+		// do not implement waddrmgr.ManagedPubKeyAddress.
+		// See RPC ImportTapscript.
+		// Fix https://github.com/lightninglabs/loop/issues/923
+		if walletAddr.Imported() {
```

Comment:
If we are sure that we don't have the internal key for any imported wallet addresses, we might as well just check `walletAddr.Imported()` before the interface check, or even earlier.

Reply:
Yeah I think we can move this check to reside before the type assertion (but after the nil check).

Reply:
I moved the check before the type assertion but after the nil check.